task/sql-tasks.js

```diff
@@ -281,7 +492,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+	        CompanyName,
+            p.ProductName,
+            d.UnitPrice as 'PricePerItem'
+        FROM Customers
+        INNER JOIN Orders o ON Customers.CustomerID = o.CustomerID
+        INNER JOIN OrderDetails d ON o.OrderID = d.OrderID
+        INNER JOIN Products p ON p.ProductID = d.ProductID
+        WHERE d.UnitPrice = (SELECT MAX(d.UnitPrice)
+            FROM Customers c
+            INNER JOIN Orders o ON Customers.CustomerID = o.CustomerID
+	        INNER JOIN OrderDetails d ON o.OrderID = d.OrderID
+            WHERE Customers.CompanyName = c.CompanyName)

```

Comment:
company name doesn't uniquely identify customer according to the schema. Also company name doens't have an index, so this search is slow